vault/build/titusvaults/sources/Vault.move

```diff
@@ -140,22 +236,27 @@ module titusvaults::Vault {
             depositor: user_addr,
             amount: coin_value,
             shares_minted: shares_to_mint,
-            round: current_round.round
+            round: round_state.current_round_id
         };
         event::emit(deposit_vault_event);
     }
 
-    public (friend) fun instant_withdraw_vault<VaultT, AssetT>(account: &signer, amount: u64) acquires Round, Vault, VaultMap {
+    // instant withdraw vault
+    public (friend) fun instant_withdraw_vault<VaultT, AssetT>(account: &signer, amount: u64) acquires RoundState, Vault, VaultMap {
         let user_addr = address_of(account);
 
+        let round_state = borrow_global_mut<RoundState>(@titusvaults);
         let vault = borrow_global_mut<Vault<VaultT, AssetT>>(@titusvaults);
-        let vault_map = borrow_global_mut<VaultMap<VaultT, AssetT>>(@titusvaults);
-        let user_shares = *smart_table::borrow(&vault_map.shares, user_addr);
+        let vault_map = borrow_global_mut<VaultMap>(@titusvaults);
+        let user_shares = *smart_table::borrow(&round_state.shares, user_addr);
 
-        let current_round = borrow_global<Round>(@titusvaults);
+        // assert we are during the deposit phase
+        let current_time = timestamp::now_microseconds();
+        let deposit_end_time = round_state.round_start_time + DEPOSIT_PHASE_DURATION;
+        assert!(current_time <= deposit_end_time, E_NOT_DEPOSIT_PHASE);
 
         let current_deposit = *smart_table::borrow(&mut vault_map.deposits, user_addr);
-        assert!(current_round.round == vault.creation_round, E_NOT_INSTANT_WITHDRAWAL);
+        assert!(round_state.current_round_id == vault.creation_round, E_NOT_INSTANT_WITHDRAWAL);
```

Comment:
no, instant withdraw is allowed between deposit start time and option_creation_time only.

Reply:
@0xriazaka 
This is based on time and not ID.